src/movies.js

```diff
@@ -1,15 +1,58 @@
+
+
 // Iteration 1: All directors? - Get the array of all directors.
+const getAllDirectors = movies.map(function(movie){
+    return(movie.director)
+  })
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-
+function howManyMovies(movies) {
+    const dramaMovies = movies.filter(function (movie) {
+      if (movie.genre.includes("Drama") === true && movie.director === "Steven Spielberg") {
+        return true;
+      }
+    });
+    return dramaMovies.length;
+}
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
-
+function ratesAverage(movies){
+    const sumRates = movies.reduce(function(acc,val){
+      return acc + val.rate
+    },0)
+    return +(sumRates / movies.length).toFixed(2)
+    
+  } 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-
+function dramaMoviesRate(movies){
+    let dramas = movies.filter(function(movie) {
```

Comment:
Both variables dramas and avg will not be reassigned, so const instead of let can be used.